### PR TITLE
Support RunEnd array with bool values

### PR DESCRIPTION
**Pull request by robert3005**
This is necessary if we want to support compute pushdown as we will create
boolean values, otherwise we have to immediately canonicalize

### CODE REVIEW

File: encodings/runend/src/array.rs [hunk 1]
```diff
@@ -52,6 +52,13 @@ impl RunEndArray {
         offset: usize,
         length: usize,
     ) -> VortexResult<Self> {
+        if !matches!(values.dtype(), &DType::Bool(_) | &DType::Primitive(_, _)) {
```
Comment (robert3005) [inline encodings/runend/src/array.rs:55]: added this assertion so we don't end up accidentally creating unspported arrays in the future

**Review by AdamGS — APPROVED** [review]

File: encodings/runend/Cargo.toml [hunk 2]
```diff
@@ -14,6 +14,7 @@ categories = { workspace = true }
 readme = { workspace = true }
 
 [dependencies]
+arrow-buffer = { workspace = true }
```
Comment (gatesn) [inline encodings/runend/Cargo.toml:17]: You shouldn't need this, I made BoolArray re-export BooleanBuffet since it's on its API

**Comment by lwwmanning** [comment]
@robert3005 we should just use RunEndBoolArray instead of doing this

**Comment by lwwmanning** [comment]
a la https://github.com/spiraldb/vortex/pull/1355

File: encodings/runend/Cargo.toml (hunk 2 above)
Comment (robert3005) [reply encodings/runend/Cargo.toml:17]: But I need the booleanbufferbuilder?

**Comment by robert3005** [comment]
@lwwmanning this is not a feature of compression but feature of compute. If I start with runendarray and compare it to a constant do I have to recompress it as runendbool?

**Comment by lwwmanning** [comment]
you don't have to recompress, but you might as well return a RunEndBoolArray! and we should implement the various compute pushdown operations on that

**Comment by robert3005** [comment]
I don’t see how to produce runendbool array here without decompression but would be happy to review a pr

**Comment by lwwmanning** [comment]
oh wait, duh, you'd have to merge runs

**Pull request merged**